### ISSUE TITLE
german language

### ISSUE THREAD

**Issue by Zacki84** [MycroftAI/mycroft-core#924]
Hi
I´m very interested in using mycroft.
Is there a way to change language from english to german?
Thanks

**Comment by testman42** [comment]
there are some attempts at making Mycroft work with other languages, but that functionality is far from being completed.

**Comment by augustnmonteiro** [comment]
@Zacki84 you can check out this pull request #814 it isn't merged yet

**Comment by eX00r** [comment]
Are there any news on this issue?

**Comment by forslund** [comment]
@eX00r we're slowly getting there. the https://translate.mycroft.ai has got a bunch of skills completely translated and we're step by step importing it. Core has gotten a some German specific code added (extracting time/date and such from german)

TTS and STT still need to be configured manually but we're getting there and people are running it using german.

**Comment by eX00r** [comment]
Thanks for your answer. Unfortunately I wasn't able to configure TTS and STT right altough I followed the guide (https://mycroft.ai/documentation/language-support/german/) step by step. 

It would be very helpful if the process is described more detailed or at least the snippets of what to change manually are bigger. Maybe a complete guide including an example of the customized 'mycroft.config' and the necessary changes to 'listener.py' and 'pocketsphinx_audio_consumer.py' is the best way.

**Comment by forslund** [comment]
If you're not on our [mattermost chat](chat.mycroft.ai) I suggest you join and ask in the language-de for specifics.

I'll ping @KathyReid about the documentation. You can also raise an issue against the [documentation repo](https://github.com/MycroftAI/docs-rewrite) and maybe the german users can lend a hand updating the documentation.

**Comment by KathyReid** [comment]
Our language support at the moment is _very experimental_, and the instructions provided are essentially a "best guess" at the moment. There is a lot that needs to be done before we can support additional languages across the entire Mycroft Voice Stack - all of the elements are documented at; 
https://mycroft.ai/documentation/language-support/adding-language/

We are currently building out our [Mycroft Translate](https://translate.mycroft.ai) platform that will help to crowdsource translation of Skill `dialog` and `vocab` files, however we still have a lot of work to do before we have a German STT (likely based on DeepSpeech). 

We are currently working on tools which will allow a German TTS to be built, however there already exist German TTS packages like `espeak`. 

In summary we will be focussing on building automated tools rather than going back and updating the experimental languages documentation.

**Comment by eX00r** [comment]
Thank you. That totally makes sense. So I guess engaging in Mycroft Translate and DeepSpeech is the best way to invest my time.

**Comment by danielwine** [comment]
@KathyReid  This could be slightly offtopic, but I would be glad if you could help me.
I just started working on the hungarian language support, so I'm not quite ready yet.
Should I create an issue for it first (as written in the contributing guidelines) or should I simply create a pull request when I'm done with my specific feature branch? Sorry for the naive question, this will be my first pull request ever.

**Comment by forslund** [comment]
A lot of german vocab and functions have been added and people are using Mycroft in German today. Since this is something of a moving target I'm going to close this issue, there are of course more work to be done but perhaps smaller more well defined issues can be created for anything missing / broken when using Mycroft in German.
